proto/requests.proto

```diff
@@ -68,6 +68,25 @@ message SyncStateRequest {
     repeated uint32 nullifiers = 4;
 }
 
+// State synchronization request.
+//
+// Specifies state updates the client is intersted in. The server will return the first block which
+// contains a note matching `note_tags` or the chain tip. And the corresponding updates to
+// `nullifiers` and `account_ids` for that block range.
+message SyncNoteRequest {
+    // Last block known by the client. The response will contain the data starting from the next block,
+    // until the 
+    fixed32 block_num = 1;
+
+    // Determines the tags which the client is interested in. These are only the 16high bits of the
+    // note's complete tag.
+    //
+    // The above means it is not possible to request an specific note, but only a "note family",
+    // this is done to increase the privacy of the client, by hiding the note's the client is
+    // intereted on.
+    repeated uint32 note_tags = 3;
```

Comment:
I don't think this comment is accurate (this applies to the state sync endpoint as well). Specifically, note tags are now always 32 bits.

This also makes me thing that we should changed `uint32` to `fixed32`.

Reply:
I can make that change but it will be a breaking on iiuc. 

Just to make sure I understand - note tags used to be the 16 high bits, but are now the 32 high bits?

Reply:
Yeah, a breaking change is fine as I think we already assume on the client that note tags are 32 bits.

At some point in the past we did take only the high 16 bits of the tags in the request (at that point, tags were 64 bits, I believe). But now we've changed this so that tags are always 32 bits and we always request things by the exact tag (so, no need to deal with prefixes any more).

A bit more info about note tags is [here](https://github.com/0xPolygonMiden/miden-base/blob/ab7e030c3df76c8e1fe06ad4e747c28dd13f8a13/objects/src/notes/note_tag.rs#L67).

Reply:
Does that not negate the privacy aspect? Or is the hope now that there are many notes with a matching tag?

Reply:
Yes, the idea is that there will be many notes matching a specific tag. And the way it currently works, when a note is directed to a specific account, only the first 16 bits of the account ID are included in the tag (the rest are set to 0). So, not much privacy if we have only 10K accounts in the system - but as soon as there are a few million, things should become quite a bit more private.